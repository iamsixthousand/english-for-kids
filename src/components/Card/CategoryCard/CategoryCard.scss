@import '../../../index.scss';

.CategoryCard {
  height: 100%;
  border-radius: 10px;
  opacity: 0.9;
  overflow: hidden;
  &:hover {
    opacity: 1;
  }
}

.CardImageContainer {
  display: flex;
  justify-content: center;
  align-items: center;
}

.CardImage {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.CategoryText {
  height: 20%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0;
  font-weight: $bold;
}
