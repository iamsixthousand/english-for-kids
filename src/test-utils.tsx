import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import gameProcessReducer from './state/gameProcessReducer';

const WithStoreProvider: React.FC = ({ children }) => {
  const myStore = createStore(
    combineReducers({
      gameProcess: gameProcessReducer,
    })
  );
  return <Provider store={myStore}>{children}</Provider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: WithStoreProvider, ...options });

export * from '@testing-library/react';
export { customRender as render };
