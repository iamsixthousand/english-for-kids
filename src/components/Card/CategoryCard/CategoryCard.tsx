import React from 'react';
import { PUBLIC_URL } from '../../../@core/constants';
import './CategoryCard.scss';

interface CategoryCardProps {
  title: string;
  image: string;
  isPlaying: boolean;
}

export const CategoryCard: React.FC<CategoryCardProps> = ({ isPlaying, title, image }) => {
  return (
    <div className="CategoryCard">
      {isPlaying}
      <div className="CardImageContainer">
        <img className="CardImage" alt={title} src={`${PUBLIC_URL}/${image}`} />
      </div>
      <div className="CategoryText">{title}</div>
    </div>
  );
};
