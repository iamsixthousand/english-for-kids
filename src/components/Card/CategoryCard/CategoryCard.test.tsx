/* eslint-disable no-unused-expressions */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react';
import { screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import { CategoryCard } from './CategoryCard';
import { categories } from '../../../cardData';
import { render } from '../../../test-utils';

describe('CategoryCard component:', () => {
  const { title } = categories[0];
  const { image } = categories[0];

  it('should contain correct input title text', () => {
    render(<CategoryCard title={title} image={image} />);
    expect(screen.getByText('Action (set A)')).toBeInTheDocument();
  });
  it('should contain image', () => {
    render(<CategoryCard title={title} image={image} />);
    expect(screen.getByRole('img')).toBeInTheDocument();
  });
  it('matches snapshot', () => {
    const categoryCard = render(<CategoryCard title={title} image={image} />);
    expect(categoryCard).toMatchSnapshot();
  });
});
