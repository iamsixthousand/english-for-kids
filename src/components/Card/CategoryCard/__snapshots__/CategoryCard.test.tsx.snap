// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard component: matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CategoryCard"
      >
        <div
          class="CardImageContainer"
        >
          <img
            alt="Action (set A)"
            class="CardImage"
            src="/img/action-set-a.jpg"
          />
        </div>
        <div
          class="CategoryText"
        >
          Action (set A)
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CategoryCard"
    >
      <div
        class="CardImageContainer"
      >
        <img
          alt="Action (set A)"
          class="CardImage"
          src="/img/action-set-a.jpg"
        />
      </div>
      <div
        class="CategoryText"
      >
        Action (set A)
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
